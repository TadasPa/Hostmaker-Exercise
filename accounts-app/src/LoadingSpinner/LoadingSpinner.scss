@import "../constants.scss";

.spinner-container {
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0px;
  left: 0px;
}

.overlay {
  position: absolute;
  width: 100%;
  height: 100%;
  background-color: #d8d8d8;
  opacity: 0.5;
}

.spinner {
  position: absolute;
  animation-name: spin;
  animation-duration: 0.8s;
  animation-iteration-count: infinite;
  animation-timing-function: linear;
  border-radius: 50%;
  border-style: solid;
  border-width: 0.5rem;
  width: 5rem;
  height: 5rem;
  border-color: $button-collor transparent $button-collor $button-collor;
  top: 50%;
  left: 50%;
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
