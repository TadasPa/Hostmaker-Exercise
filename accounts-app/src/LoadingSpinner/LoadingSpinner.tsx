import React, { Component } from "react";
import "./LoadingSpinner.scss";

export interface ILoadingSpinnerProps {
  isLoading: boolean;
}

class LoadingSpinner extends Component<ILoadingSpinnerProps> {

  render() {
    if (this.props.isLoading) {
      return (
        <div className="spinner-container">
          <div className="overlay" />
          <div className="spinner" />
        </div>
      ); 
    } else {
      return null;
    }      
  }
}

export default LoadingSpinner; 