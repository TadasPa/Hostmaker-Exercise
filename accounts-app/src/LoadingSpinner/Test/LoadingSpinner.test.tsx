import React from 'react';
import { shallow } from 'enzyme';
import { LoadingSpinner } from '../LoadingSpinner';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  shallow(<LoadingSpinner isLoading={false} />);
});


test('LoadingSpinner renders correctly when not loading', () => {
  const spinner = renderer
    .create(<LoadingSpinner isLoading={false} />)
    .toJSON();
  expect(spinner).toMatchSnapshot();
});

test('LoadingSpinner renders correctly when loading', () => {
  const spinner = renderer
    .create(<LoadingSpinner isLoading={true} />)
    .toJSON();
  expect(spinner).toMatchSnapshot();
});