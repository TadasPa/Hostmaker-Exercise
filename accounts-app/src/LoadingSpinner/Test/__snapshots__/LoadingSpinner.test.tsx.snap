// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner renders correctly when loading 1`] = `
<div
  className="spinner-container"
>
  <div
    className="overlay"
  />
  <div
    className="spinner"
  />
</div>
`;

exports[`LoadingSpinner renders correctly when not loading 1`] = `null`;
